### PR TITLE
Upgrade node 16 + fix CD + npm ci

### LINKED ISSUE #202: Upgrade Node Version
We're currently using Node 12 ( at least according to CI) which is currently in maintenance LTS.
We should migrate to at least Node 14, I advocate we go to 16 though.

image

### PR DESCRIPTION
**Pull request by mkhraisha**
This PR fixes #202 #207 and modifies our `npm install` to `npm ci` 

This PR will not work as is because of the issues outlined in #207 , we need to create a service user to push to protected branches.

Changes:
* Modify `npm i` `npm install` to `npm ci`. This allows us to install from `package.lock` rather than resolving dependencies on the fly.
* Adds a requirement for `ACTION_AUTH_TOKEN ` to be defined in the secrets of the repo, which should correspond to a PAT for a user with admin privileges that can bypass the protected branch issue. [community](https://github.community/t/how-to-push-to-protected-branches-in-a-github-action/16101/4h)
* Migrates to Node 16 from Node 12.

DO NOT MERGE WITHOUT SETTING ACTION_AUTH_TOKEN

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -34,13 +36,14 @@ jobs:
         uses: 'phips28/gh-action-bump-version@master'
         with:
           tag-prefix: 'v'
+          commit-message: 'CI: bumps version to {{version}} [skip ci]'
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.ACTION_AUTH_TOKEN }}
```
Comment (OR13) [inline .github/workflows/cd.yml:41]: why is the necssary? can we just use this? 

https://github.com/decentralized-identity/JWS-Test-Suite/blob/main/.github/workflows/ci.yml#L32

We should not need secrets for this IMO, the default github tokens should be enough.

Comment (mkhraisha) [reply .github/workflows/cd.yml:41]: The default github tokens cannot push to protected branches

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -34,13 +36,14 @@ jobs:
...
 
       - name: Deploy test report
         uses: peaceiris/actions-gh-pages@v3
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_token: ${{ secrets.ACTION_AUTH_TOKEN }}
```
Comment (OR13) [inline .github/workflows/cd.yml:46]: please use the secret available to all actions, we don't have branch protection on, so it should be sufficient.

Comment (mkhraisha) [reply .github/workflows/cd.yml:46]: Will update!

**Comment by mkhraisha** [comment]
Need to update the CI to be used on Push and on PR

**Comment by mkhraisha** [comment]
This should fix the CI issues and it shoudl be ready now.

**Review by mprorock — APPROVED** [review]

**Comment by mprorock** [comment]
@OR13 can you review

**Pull request merged**